File: C++/Builder/LaptopManufacturer/laptop_manufacturer.cc
#include "laptop_manufacturer.h"

LaptopManufacturer::LaptopManufacturer(LaptopBuilder *builder) : _builder(builder) {}

void LaptopManufacturer::set_builder(LaptopBuilder *builder) {
    _builder = builder;
}

void LaptopManufacturer::build_laptop() {
    _builder->set_laptop_name();
    _builder->install_cpu();
    _builder->install_memory();
    _builder->install_os();
}

std::unique_ptr<Laptop> LaptopManufacturer::deliver_laptop() {
    return _builder->get_laptop();
}

File: Java/Builder/LaptopManufacturer/src/com/ericgtkb/LaptopManufacturer.java
package com.ericgtkb;

public class LaptopManufacturer {
    private LaptopBuilder builder;

    public LaptopManufacturer(LaptopBuilder builder) {
        this.builder = builder;
    }

    public void setBuilder(LaptopBuilder builder) {
        this.builder = builder;
    }

    public void buildLaptop() {
        builder.setLaptopName();
        builder.installCpu();
        builder.installMemory();
        builder.installOs();
    }

    public Laptop deliverLaptop() {
        return builder.getLaptop();
    }
}
